File: cash/cash.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int change;
    do
    {
        // Prompt user for the change owed
        change = get_int("Change: ");
    }
    while (change < 0);

    // Declare a variable to store the number of coins
    int counter = 0;
    do
    {
        // If the change is greater than 25, the program subtracts 25 and add one to the counter
        if (change >= 25)
        {
            change -= 25;
            counter++;
        }
        else if (10 <= change && change < 25)
        {
            change -= 10;
            counter++;
        }
        else if (5 <= change && change < 10)
        {
            change -= 5;
            counter++;
        }
        else if (change < 5 && change > 0)
        {
            change -= 1;
            counter++;
        }
    }
    // Repeat the process
    while (change > 0);

    // Print the number of coins
    printf("%i\n", counter);
}

File: sentimental-cash/cash.py
from cs50 import get_float

# Prompt the user for the change owed
change = get_float("Change: ")

# If the float is not valid, continue to promt the user
while change < 0:
    change = get_float("Change: ")

# I convert dollars to cents (rounding to avoid the imprecision
# of the floating value)
change = int(round(change * 100, 1))
# Initialize the number of coins to zero
coins = 0

# Repeat the process until change is equal to 0
while change > 0:
    if change >= 25:
        # I add to the number of coins the ratio
        # between the change owed and the coin
        coins = coins + (change // 25)
        change = change - (change // 25) * 25
    elif change >= 10:
        coins = coins + (change // 10)
        change = change - (change // 10) * 10
    elif change >= 5:
        coins = coins + (change // 5)
        change = change - (change // 5) * 5
    else:
        coins = coins + (change // 1)
        change = change - (change // 1) * 1

# Print the number of coins
print(f"{coins}")
